Broccoli test.js.cjs

IgorMinar opened the pull request.

IgorMinar committed:
chore(build): refactor test.unit.cjs to use the broccoli pipeline

This change solves several problems:
- the broccoli pipeline is used to compile the node/cjs tree upon any change to the modules/ directory
- jasmine tests run in a new process removing the need to clean up environment after each test
- since we transpile only those test files that are actually needed for node/cjs build, we transpile less and don't need to filter out tests

IgorMinar commented:
benchpress is unhappy with me because it can't find di which is because I removed the node_modules hack.

I'll need to go back make sure that benchpress has the right NODE_PATH set when starting.... tomorrow.

IgorMinar commented:
I reverted the part that was setting the NODE_PATH. for some reason I can't control NODE_PATH on travis. We might come back to this later when other more pressing issues are resolved.

I'm going to wait with merging this in until we have the incremental compilation resolved. Right now this change makes the cjs workflow pretty painful.

IgorMinar committed:
build(broccoli): add tree-differ for diffing broccoli trees

IgorMinar committed:
build(broccoli): add custom broccoli.d.ts file

IgorMinar committed:
chore(build): add tsd.cached.json to our .gitignore

IgorMinar committed:
build(broccoli): add testing infrastructure for our build plugins

components:
- gulp test.unit.broccoli task
- mock-fs for mocking our FS in unit tests
- jasmine d.ts file for type checking

jasmine lib is provided by minijasmine2 so we don't need to include it explicitly

IgorMinar committed:
build(broccoli): convert traceur and broccoli-dest-copy plugins to use tree-differ

Also adding symlink-or-copy to our npm dependencies since our plugins now use it.

alexeagle commented:
got through all the files, looking good. Leaving typescript compiler as a follow-up task?

IgorMinar committed:
build(broccoli): add source-maps to our broccoli tools

IgorMinar committed:
docs(npm-shrinkwrap.readme.md): update instructions

IgorMinar committed:
build(gulp): refactor test.unit.broccoli to use task dependencies and not die when a compilation error occurs

With this change we also stop gulp task queue when an error in build.broccoli.tools occurs, that way we
don't bother testing when transpilation fails.

IgorMinar commented:
wrt the typescript plugin: yeah, I want to get this in and then we can work on more plugins. I'll also likely refactor how the differ is used to reduce the boilerplate. but let's get this in first.

IgorMinar committed:
chore(build): exclude tree-differ.ts from format-checking due to a bug in clang-format

see https://github.com/angular/clang-format/issues/4

IgorMinar committed:
build(gulp): use strict mode for gulpfile

- a variable `public` was renamed since it is a reserved keyword in the strict mode

IgorMinar committed:
refactor(build): introduce AngularBuilder facade and clean up many things

IgorMinar committed:
build(gulp): rename tasks `build.broccoli.tools` => `build.tools` + `test.unit.broccoli` => `test.unit.tools`

IgorMinar committed:
build(gulp): remove obsolete gulp transformer

IgorMinar committed:
build(gulp): remove gulp-jasmine, run tests in a new process instead

otherwise we see occasional failures due to require.cache and other global state collisions

IgorMinar commented:
@alexeagle I made some major changes to unify the ci/development tasks for cjs builds. PTAL from 0980885

IgorMinar committed:
build(gulp): remove obsolete config

IgorMinar committed:
build(js.cjs): re-enable tests that now pass in cjs

previously these tests were failing but now they pass on master

IgorMinar committed:
build(gulp): add build/clean.tools task prevent test.unit.tools/ci from building

on our CI server we currently split each build into the building phase and testing phase, this change aligns test.unit.tools/ci
with the rest of ci test taskswq

IgorMinar committed:
build(gulp): add test.unit.tools to test.js task

IgorMinar committed:
build(gulp): fix incremental compilation by reusing angularBuilder across watch re-reruns

alexeagle commented on gulpfile.js:
```diff
@@ -582,28 +465,41 @@ gulp.task('build.dart', function(done) {
 });
 
+
 gulp.task('build.js.dev', function(done) {
```
inline and remove the broccoli.js.dev task (should be symmetric w/ prod)

IgorMinar replied:
I'd love to but I need it elsewhere. I still need to refactor build.js.dev and build.js.prod tasks.

alexeagle commented on tools/broccoli/broccoli-dest-copy.ts:
```diff
@@ -1,33 +1,63 @@
-  getDestFilePath(relativePath: string): string { return relativePath; }
 
-  processString(content: string, relativePath: string): string { return content; }
+class DestCopy implements BroccoliTree {
```
it's strange that the 'Tree" means both a particular intermediate filesystem as well as a transformation between two of them

IgorMinar replied:
yeah. I know. there are at least 3 things that are being referred to as "tree" in broccoli. (broccoli plugin, fs directory tree, string pointing to the fs directory tree).

for now I'm pretending that this issue doesn't exist.

alexeagle commented on tools/broccoli/traceur/index.ts:
```diff
@@ -1,55 +1,95 @@
+/// <reference path="../broccoli.d.ts" />
```
I recall one reason for `traceur/index.js` rather than `traceur.js` was that we forked these from the gulp equivalent, do we still want to acknowledge that somehow?

IgorMinar replied:
that was several refactorings ago. At this point there is nothing left of of the original code and I still plan on creating a better abstraction for our plugins (something like `BroccoliDiffingWriter` but without the use of inheritance), so this file will still change quite a bit.

IgorMinar commented:
I'm looking into the test failure now

IgorMinar commented:
found the issue. we were running `test.unit.tools` rather than `test.unit.tools/ci` on ci which started watching fs and waited causing timeout. fixed now.

alexeagle commented:
Great, looking forward to getting this and the Karma change in. I'm working
on getting the ES6 emit to work, so we can go pure-TypeScript (for a couple
directories)

On Mon, Apr 27, 2015 at 10:13 AM Igor Minar <notifications@github.com>
wrote:

> found the issue. we were running test.unit.tools rather than
> test.unit.tools/ci on ci which started watching fs and waited causing
> timeout. fixed now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular/pull/1356#issuecomment-96743940>.
>

IgorMinar commented:
landed as 725f909ff82cbb30754603c0ee9e489fa8c4a336...8ea03d0380654f745b67e9602c88164fad684e8d
